packages/@aws-cdk/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-scaling-policy.ts

```diff
@@ -0,0 +1,28 @@
+import * as path from 'path';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import { App, Stack } from '@aws-cdk/core';
+import * as integ from '@aws-cdk/integ-tests';
+import { QueueProcessingFargateService } from '../../lib';
+
+const app = new App();
+const stack = new Stack(app, 'aws-ecs-patterns-queue-scaling-policy');
+const vpc = new ec2.Vpc(stack, 'VPC', {
+  maxAzs: 2,
+});
+
+new QueueProcessingFargateService(stack, 'QueueProcessingService', {
+  vpc,
+  memoryLimitMiB: 512,
+  image: new ecs.AssetImage(path.join(__dirname, '..', 'sqs-reader')),
+  minScalingCapacity: 0,
+  scalingSteps: [{ upper: 0, change: -1 }, { lower: 5, change: +1 }, { lower: 10, change: +5 }],
+  enableCpuBasedScaling: true,
+  cpuBasedScalingTargetUtilization: 30,
+});
+
+new integ.IntegTest(app, 'queueProcessingFargateServiceTest', {
+  testCases: [stack],
+});
+
+app.synth();
```

Comment:
integ test is fine but can we get a unit test for this prop funcitonality?

Reply:
sure, i don't see why not